ivy_tests/test_ivy/test_frontends/test_torch/test_nn/test_functional/test_pooling_functions.py

```diff
@@ -173,8 +166,12 @@ def test_torch_avg_pool3d(
 ):
     input_dtype, x, kernel_size, stride, padding = dtype_x_k_s
 
-    padding = [padding[i][0] for i in range(len(padding))]
-
+    if len(stride) == 1:
+        stride = [stride[0]] * 3
+    x_shape = x[0].shape[2:]
+    padding = (pad[i] for i, pad in enumerate(padding))
+    if not sum(padding) == 0:
+        padding = calculate_same_padding(kernel_size, [stride[0]] * 3, x_shape)
```

Comment:
I don't think we need to process the padding returned by `arrays_for_pooling`. We have set `only_explicit_padding=True` so we are going to receive a tuple of integers as padding, which matches what the torch function expects.
Same for the rest of the tests above.

Reply:
If that was needed because our frontend function currently fails for some of the cases that should be supported, you can leave this in the test but add a ToDo comment to remove that once the ivy functions have been updated to allow the frontend to handle all possible explicit padding cases.

Reply:
I broadcast the kernel_size param for the padding specification below it's essential for specifying the padding algorithm `SAME` or `VALID` until we update the current implementation of the pooling functions to accept also explicit padding we have to do so I think?

Reply:
@AnnaTz 
![image](https://github.com/unifyai/ivy/assets/97109954/6374069c-33da-4ab4-aa51-b527a2eb9d1b)
I hope my explanation make sense if not 😕 you can see from the above image if I take the padding from array for pooling directly without processing it will give me this error

Reply:
> If that was needed because our frontend function currently fails for some of the cases that should be supported, you can leave this in the test but add a ToDo comment to remove it once the ivy functions have been updated to allow the frontend to handle all possible explicit padding cases.
yeah, indeed it's related. sorry for the two comments above I wrote them cuz i didn't see your last comment